### PR TITLE
Add in a GitHub workflow.

### PR DESCRIPTION
**Pull request by clalancette**
Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

Draft until I verify it actually works.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    - name: Setup python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        architecture: x64
+
+    - name: Install pip dependencies
+      run: |
+       python -m pip install flake8 flake8-blind-except flake8-builtins flake8-class-newline flake8-comprehensions flake8-deprecated flake8-docstrings flake8-import-order flake8-quotes pytest
+       python -m pip freeze
```
Comment (hidmic) [inline .github/workflows/test.yml:32]: @clalancette maybe cache pip installed packages to speed things up a bit? See [actions/cache@v2](https://github.com/actions/cache/blob/main/examples.md#python---pip).

**Review by hidmic — COMMENTED** [review]
Overall makes sense to me.

File: .github/workflows/test.yml (hunk 1 above)
Comment (aprotyas) [inline .github/workflows/test.yml:32]: Just curious, what does this achieve?

Also, I know this is typically handled with `rosdep` but for this workflow's benefit, would a `requirements.txt` file containing the `pip freeze` for this project be a good idea?

Comment (clalancette) [reply .github/workflows/test.yml:32]: The idea with the `pip freeze` is really just to print out the versions of all of the pip packages are being installed.  It's useful if an error occurs and you are trying to recreate the environment locally.

`pip list` would do as well, but we tend to prefer `pip freeze` in our infrastructure for reasons I don't fully understand.

Comment (clalancette) [reply .github/workflows/test.yml]: Done in 7f8d743

Comment (clalancette) [reply .github/workflows/test.yml:39]: Switching to requirements.txt is a good idea.  I've done this in 7f8d743.

**Review by hidmic — APPROVED** [review]
LGTM

**Comment by clalancette** [comment]
Thanks!  Merging.

**Pull request merged**